DirectEthernet/src/main/scala/org/labrad/ethernet/server/DirectEthernet.scala

```diff
@@ -190,15 +201,7 @@ extends ServerContext with Logging {
             |received packet as a byte string""")
   def read(numPackets: Long = 1): Seq[(String, String, Int, Array[Byte])] = {
     collect(numPackets)
-    val results = queue.take(numPackets.toInt)
-    results.map { pkt =>
-      val src = pkt.getHeader.getSrcAddr
-      val dst = pkt.getHeader.getDstAddr
-      val typ = pkt.getHeader.getType.value.toInt
-      val data = pkt.getPayload.getRawData
-
-      (src.toString.toUpperCase, dst.toString.toUpperCase, typ, data)
-    }
+    processPackets(queue.take(numPackets.toInt))
```

Comment:
If you are asking for more that 2 billion packets, you have bigger problems. We could add a check, but I'm not too worried about it. Using `Long` as a proxy for `unsigned 32-bit int` is dumb, but it's going to take a lot of work to fix that.